apps/lasp/src/lasp_core.erl

```diff
@@ -337,6 +337,26 @@ bind(Id, Value, Store) ->
 
 %% @doc Define a dataflow variable to be bound a value.
 -spec bind(id(), value(), function(), store()) -> {ok, var()}.
+bind(Id, {delta, Value}, MetadataFun, Store) ->
+    Mutator = fun(#dv{type=Type, metadata=Metadata0, value=Value0,
+                      waiting_threads=WT}=Object) ->
+            Metadata = MetadataFun(Metadata0),
+            %% Merge may throw for invalid types.
+            try
+                Merged = lasp_type:merge(Type, Value0, Value),
+                {ok, SW} = reply_to_all(WT, [], {ok, {Id, Type, Metadata, Merged}}),
+                NewObject = #dv{type=Type, metadata=Metadata, value=Merged, waiting_threads=SW},
+                %% Return value is a delta state
+                {NewObject, {ok, {Id, Type, Metadata, Value}}}
```

Comment:
bind() with {delta, Value} returns the delta state (Value) not the entire state (Merged). update() for the broadcast backend is using the state twice (one for the broadcast and the other for the return value). For the delta-mutation, the broadcast should use the delta state, but I am not sure about the return value of update(). Which state is the right one for the return value of update() with the delta-mutation (the delta state or the entire state)?